content/blog/scheduling-in-react/index.md

```diff
@@ -0,0 +1,237 @@
+
+  At the time of writing this article, the priority levels are:
+
+  - `Immediate` for tasks that need to run synchronously.
+  - `UserBlocking` (250ms timeout) for tasks that should run as the result of a user interaction (e.g. a button click).
+  - `Normal` (5s timeout) for rendering non-interactive parts of the screen.
+  - `Low` (10s timeout) for tasks that can be deferred but must still complete eventually (e.g. an analytics ping).
+  - `Idle` (no timeout) for tasks that do not have to run at all. (e.g. hidden off-screen content).
+
+  The timeouts for each priority level are necessary to make sure that lower priority work still runs even if we have so much higher priority work to do, that the higher priority work could run continuously. In scheduling algorithms, this problem is referred to as [starvation](<https://en.wikipedia.org/wiki/Starvation_(computer_science)>). The timeouts give us the guarantee that every scheduled task will eventually run. For example we won't miss a single analytics ping even if we have ongoing animations in our app.
+
+  ➡️ With this feature, we're can run tasks with different importance.
+
+## Scheduling in Action
+
+Let's see how we can use these features to make an app feel a lot more responsive. To do this, we take a look at [ScheduleTron 3000](https://github.com/philipp-spiess/scheduletron3000), an app that I built that allows us to highlight a search term in a list of names. Let's take a look at the implementation first:
+
+<!-- prettier-ignore -->
+```js
+function App() {
+  const [searchValue, setSearchValue] = React.useState();
+
+  function handleChange(value) {
+    setSearchValue(value);
+  }
+
+  return (
+    <div>
+      <SearchBox onChange={handleChange} />
+      <NameList searchValue={searchValue} />
+    </div>
+  );
+}
+
+function SearchBox(props) {
+  const [inputValue, setInputValue] = React.useState();
+
+  function handleChange(event) {
+    const value = event.target.value;
+
+    setInputValue(value);
+    props.onChange(value);
+    sendAnalyticsPing(value);
+  };
+
+  return (
+    <input
+      type="text"
+      value={inputValue}
+      onChange={handleChange}
+    />
+  );
+}
+
+ReactDOM.render(<App />, container);
+```
+
+_ℹ️ This example uses [React Hooks](https://reactjs.org/docs/hooks-intro.html). If you're not familiar with them yet, take a look at the [CodeSandbox code](https://codesandbox.io/s/j3zrqpzkr5). Addition you might wonder why we use two different state variables for this example. We'll find out why in a bit._
+
+The `<SearchBox>` displays the input element which is controlled by the local state variable `inputValue`. When the input changes, it will call the `onChange` prop with the new value. The new value is then stored inside the `<App>` component's `searchValue` state and passed over to the `<NameList>`. At the end, we also simulate an analytics ping to our servers.
```

Comment:
Maybe add this explanations directly to the code as comments so it's easier to connect? (Not 100% sure if it makes sense though)